plugin/pkg/shoot/validator/admission.go

```diff
@@ -226,6 +237,11 @@ func (v *ValidateShoot) Admit(ctx context.Context, a admission.Attributes, o adm
 			return admission.NewForbidden(a, fmt.Errorf("cannot create shoot '%s' in project '%s' already marked for deletion", shoot.Name, project.Name))
 		}
 		addInfrastructureDeploymentTask(shoot)
+	case admission.Delete:
+		if shoot.Status.LastOperation != nil && (shoot.Status.LastOperation.Type == core.LastOperationTypeRestore ||
+			shoot.Status.LastOperation.Type == core.LastOperationTypeMigrate) {
+			return admission.NewForbidden(a, fmt.Errorf("cannot mark shoot %s for deletion during %s operation", shoot.Name, shoot.Status.LastOperation.Type))
```

Comment:
```suggestion
			return admission.NewForbidden(a, fmt.Errorf("cannot mark shoot %s/%s for deletion during %s operation", shoot.Namespace, shoot.Name, shoot.Status.LastOperation.Type))
```

It is always a good idea to add the obj namespace as common obj names can be shared across namespaces and someone may wonder exactly which obj is causing the error.